### ISSUE TITLE
Bodhi should display which required tests are missing

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#2227]
@ralphbean [noted on Fedora's devel list](https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/B7SLNO2H3PGLMKHMDNSATU5HGUYISLRK/) that Greenwave does give a list of missing tests to Bodhi when it blocks an update due to missing or no tests. We should display the names of the missing tests to users.

**Comment by sharkcz** [comment]
thx, I'm missing exactly this feature in the current web UI

**Comment by ralphbean** [comment]
Yeah!  We can take it one step further too.  If Bodhi passes `verbose=True` to Greenwave in the original question, it will get back *all* the results that Greenwave sees in resultsdb, *including* the ones not required by the Greenwave policy.  Bodhi could then cache and display these on the Test Results page, which would *greatly* reduce the load on resultsdb itself.  Today, (I think), Bodhi presents javascript that, when executed in users' browsers causes them to fetch results from resultsdb.  I don't have data from resultsdb's apache logs, but I bet this has a "thundering herd" effect when more than a few people are using Bodhi at the same time.

**Comment by bowlofeggs** [comment]
@ralphbean Yeah I'm becoming convinced that it would be good for Bodhi to cache that stuff. Plus I can use it as an excuse to kill some JS and bring that stuff back into my happy place in Python land (with tests!). Well, maybe it'll just be template land, but still!

**Comment by sharkcz** [comment]
https://pagure.io/greenwave/issue/124 to provide both satisfied and unsatisfied requirement already exists

**Comment by ralphbean** [comment]
Thanks for digging in @sharkcz.  That one should've been closed a while ago, with https://pagure.io/greenwave/pull-request/125
